### ISSUE TITLE
Not every injected client is disposable

### ISSUE THREAD

**Issue by dltnbs** [RicoSuter/NSwag#3297]
I use `HttpClientType` to allow me to inject my own client type, which is not disposable. This was fine, because with `InjectHttpClient` defaulting to `true`, NSwag did not emit a `Dispose` call.

I recently had to update NSwag to solve another issue and as of 13.7.1+ (#3016) it now emits a `Dispose` call that, for my client, fails to compile.

Even if, as per #3112, these calls have to be emitted despite being unreachable, could they at least be protected by an `is` and a cast?

**Comment by jeremyVignelles** [comment]
I don't know why I introduced a `disposeClient_` variable... it doesn't seem to be modified by the code, but I'm only looking on github.

https://github.com/RicoSuter/NSwag/pull/3016/files#diff-65b33706483249f53a8138451372e3d3276b904d815dcb71d0ed0c8786ea2e70R342 : this should be a liquid if instead of a C# one if that's unused...

**Comment by RicoSuter** [comment]
Yup, I think it was introduced for a simpler template - but I agree if it's always false (probably when InjectHttpClient = true) then we should not generate that at all...

**Comment by garcipat** [comment]
Can the generated clients habe IDisposable and dispose the underlying client? Or is the underlying client disposed automaticly?
